### PR TITLE
Custom Endpoint for categories

### PR DESCRIPTION
**Pull request by eduardogomez97**
Required by: https://github.com/doofinder/support/issues/2879

I have built the endpoint to return struct exactly the same as the normal endpoint. And url_full in extension attributes.

image

### CODE REVIEW

File: Model/CategoryListRepository.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        FilterBuilder $filterBuilder
+    ) {
+        $this->categoryRepository = $categoryRepository;
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+        $this->filterBuilder = $filterBuilder;
+    }
+
+    public function getList(SearchCriteriaInterface $searchCriteria)
+    {
+        return  $this->categoryRepository->getList($searchCriteria);
```
Comment (sofia-doofinder) [inline Model/CategoryListRepository.php:29]: no deberíamos aquí modificar el link para que se construya como esperamos en vez de el que estamos recibiendo ahora mismo? o con este endpoint el link ya viene corregido?

**Review by sofia-doofinder — CHANGES_REQUESTED** [review]
Acuerdate de subir la versión please!

**Review by davidmolinacano — APPROVED** [review]
This is absolutely awesome!!!!!! It's the best solution in my opinion!! Excellent job!!! :trophy: :medal_military: :1st_place_medal: :100: :superhero_man:

**Review by sofia-doofinder — APPROVED** [review]

**Review by ogomezba — APPROVED** [review]
LGTM!

**Pull request merged**